catalogue_of_life/Loader.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    private static boolean isValidMarineRegionID(String areaID) {
+        return isNumeric(areaID);
+    }
+
+    private static boolean isNumeric(String s) {
+        try {
+            Long.parseLong(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void loadData(Path dataDirectory, Options options) throws IOException {
+        extractSchema(dataDirectory);
+
+
+        LoadOptions loadOptions = new LoadOptions();
+        loadOptions.dataConfigFilePath = createTypeDBLoaderConfig(dataDirectory).toString();
+        loadOptions.databaseName = "catalogue-of-life";
+        loadOptions.typedbURI = options.typedbURI();
+        loadOptions.cleanMigration = true;
+        loadOptions.loadSchema = false;
+
+        TypeDBLoader loader = new TypeDBLoader(loadOptions);
+        loader.load();
+    }
+
+    public static void extractSchema(Path dataDirectory) throws IOException {
+        InputStream schema = Loader.class.getResourceAsStream("/catalogue_of_life/schema.tql");
+        if (schema == null) throw new FileNotFoundException("schema.tql not found within the jar!");
+        Files.copy(schema, dataDirectory.resolve("schema.tql"), StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    public static Path createTypeDBLoaderConfig(Path dataDirectory) throws IOException {
+        Path config = dataDirectory.resolve("loader-config.json");
+
+        try (FileWriter configWriter = new FileWriter(config.toFile())) {
+            InputStream template = Loader.class.getResourceAsStream("/catalogue_of_life/loader-config.json.template");
+            if (template == null) throw new FileNotFoundException("loader-config.json.template not found within the jar!");
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(template));
+            String replacement = dataDirectory.toString();
+            while (reader.ready()) {
+                String line = reader.readLine().replace("${data-directory}", replacement);
+                configWriter.write(line + "\n");
+            }
+        }
+
+        return config;
+    }
+
+    @CommandLine.Command(name = "loader", mixinStandardHelpOptions = true)
+    public static class Options {
+        @CommandLine.Option(
+                names = {"--working-directory", "--dir"},
+                defaultValue = ".",
+                description = "directory in which to store data (default: current)"
+        )
+        private String workingDirectory;
+
+        @CommandLine.Option(
+                names = {"--skip-download"},
+                description = "do not download the dataset .zip if missing"
+        )
+        private boolean skipDownload;
+
+        @CommandLine.Option(
+                names = {"--force-download"},
+                description = "redownload the dataset .zip even if present (overrides --skip-download, implies --force-unzip)"
```

Comment:
We keep all this override and impication complexity confined to this class so the code above doesn't need to be aware of it.